### ISSUE TITLE
Freezing igspoker.cpp games

### ISSUE THREAD

**Issue by Cristiano-MDQ** [mamedev/mame#4183]
The IGS poker games like cpker, cpokerx, cpokert, have an error that when giving twice Hair Pair the game hangs and freezes and does not work anymore, being necessary to reset the nvram. Does anyone have any idea if it is an IGS encryption error or if there is a solution?

**Comment by MooglyGuy** [comment]
Based on your previous issue report as well as this one, can you please clarify your intended use case for MAME in the context of gambling games?

**Comment by Cristiano-MDQ** [comment]
I am a PCBS collector of games of chance, mainly video poker and I have tested some games dumped MAME and many still have errors. In IGS poker it is a problem to prevent the continuity of the game, if every two or three plays have to clean the nvram, all the fun of the game is lost and becomes frustrating. As for the Humans game from Subsino2.cpp, my interest is that I have evicted almost all my PCBS and then decided to test and take the first steps in the area of ​​emulation, I was able to successfully adjust everything wrong that was stopping the game and I also added a new game based on this humlan, a game subsino called Super Queen Bee, everything runs smoothly and without errors, graphics, colors, speed and settings, errors that humlan's possessed, but it is also frustrating not to see the game running 100% for not having sound and also unloading of coins. But if this is the wrong place to ask for help, I'll leave my apologies here. And if any member also likes gambling as much as I do, I will be happy to partner in the emulation of new games and I can send files dumped from some PCBS.

**Comment by MooglyGuy** [comment]
No worries! It's just unusual to encounter a user who is actually interested in gambling games. Most people seem to complain about them being in MAME. Thank you for the bug reports, hopefully someone will look into them soon! 😄

**Comment by Cristiano-MDQ** [comment]
Yes, you are right, these games have different values and you will need to map them (I'll try to do that today). But should these games have the status of NOT WORKING since they can not be played for more than 10 seconds? Thank you very much for the help, Haze.

**Comment by DavidHaywood** [comment]
It was probably just incorrectly assumed that because they look like clones of others games that the tables etc. would be the same.  They probably shouldn't currently be marked as working, no.

**Comment by angelosa** [comment]
Fixed by Robbie in 0.214 https://github.com/mamedev/mame/commit/6fc14b90edf76c6fff6d0d4ab43586b6e824a5c8#diff-2569e04fd0e421c8a816d7850ef9b0561859000d377330b48aef3509dfeed17e

**Comment by angelosa** [comment]
After quick discussion with @RobertoFresca looks like it happens when you get a double **Red Kings Pair**, my bad for not reading requirements properly here so reopening.
Driver also sports a plethora of [ROM patches](https://github.com/mamedev/mame/blob/master/src/mame/drivers/igspoker.cpp#L2122), a demotion to _MACHINE_UNEMULATED_PROTECTION_ to the affected set is probably a good first move here.
